state/protocol/badger/state.go

```diff
@@ -33,6 +33,10 @@ type State struct {
 	}
 	// cache the root height because it cannot change over the lifecycle of a protocol state instance
 	rootHeight uint64
+	// cache the spork root block height because it cannot change over the lifecycle of a protocol state instance
+	sporkRootBlockHeight uint64
```

Comment:
The intention behind [my suggestion to introduce `sporkRootBlockHeight`](https://github.com/dapperlabs/flow-go/issues/6443#issuecomment-1364320554) was that this variable would hold the height of the spork's root block. 
* The local history in a node might not reach all the way back to the spork's root block
* Nevertheless, the node knows the height of the spork's root block

_Suggestion_:
* I would suggest to include a getter method `SporkRootBlockHeight() (uint, error)` into the https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/params.go#L43 interface, which is exposed by `Snapshot`: https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/snapshot.go#L123
* include `sporkRootBlockHeight` into the `EncodableSnapshot`, which we use for bootstrapping. I would place `sporkRootBlockHeight` into the `Params` https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/inmem/encodable.go#L19 section
   * we assign `root.Header.Height` to `sporkRootBlockHeight` here: https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/inmem/convert.go#L290-L295
* When bootstrapping the state, we would use the value returned by the root snapshot here: https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/badger/state.go#L261
* For consensus nodes, it would be great to include an additional sanity check that the sealing segment covers the history (see condition (i) here: https://github.com/dapperlabs/flow-go/issues/6443#issuecomment-1364320554
   ```golang
   limit = max(header.Height - m.cfg.transactionExpiry, SporkRootBlockHeight)
   ```
* replace the [`IsSporkRootSnapshot`](https://github.com/onflow/flow-go/blob/a848093dccc16618a51a9d16d0b6faece2a6234b/state/protocol/util.go#L67-L77) with a direct comparison of `root